### PR TITLE
[`Core generation`] Let's be less restrictive on the arguments passed to the generation calls. 

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Updates generate calls

### CODE REVIEW

File: src/transformers/generation/utils.py [hunk 1]
```diff
@@ -1953,12 +1953,13 @@ def contrastive_search(
             if model_kwargs.get("past_key_values") is None:
                 # prepare inputs
                 model_kwargs["use_cache"] = True
+                model_kwargs["output_attentions"] = output_attentions
```
Comment (gante) [inline src/transformers/generation/utils.py:1956]:
```suggestion
                self.config.output_attentions = output_attentions
```

Comment (ArthurZucker) [inline src/transformers/generation/utils.py:1956]:
```suggestion
                self.config.output_attentions = output_attentions
```

**Review by ArthurZucker — COMMENTED** [review]
slipped through the cracks

File: src/transformers/generation/utils.py [hunk 2]
```diff
@@ -2397,17 +2398,14 @@ def greedy_search(
                 # did all peers finish? the reduced sum will be 0.0 then
                 if this_peer_finished_flag.item() == 0.0:
                     break
-
-            # prepare model inputs
-            model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
+            
+            self.config.output_attentions = output_attentions
+            self.config.output_hidden_states = output_hidden_states
```
Comment (gante) [inline src/transformers/generation/utils.py:2403]: Is this one also needed?

**Review by gante — APPROVED** [review]
In general LGTM

File: src/transformers/generation/utils.py (hunk 2 above)
Comment (ArthurZucker) [reply src/transformers/generation/utils.py:2403]: anywhere we hard code `output_attentions=output_attentions,output_hidden_states=output_hidden_states` IMO

**Pull request closed without merging**